pystiche/papers/ulyanov_et_al_2016/loss.py

```diff
@@ -0,0 +1,165 @@
+from typing import Union, Optional, Sequence, Any, Dict
+from collections import OrderedDict
+import torch
+from pystiche.enc import Encoder, MultiLayerEncoder
+from pystiche.ops import (
+    MSEEncodingOperator,
+    GramOperator,
+    MultiLayerEncodingOperator,
+)
+from torch.nn.functional import mse_loss
+from pystiche.loss import MultiOperatorLoss
+from .utils import ulyanov_et_al_2016_multi_layer_encoder
+
+
+class UlyanovEtAl2016MSEEncodingOperator(MSEEncodingOperator):
+    def __init__(
+        self, encoder: Encoder, score_weight: float = 1e0
+    ):
+        super().__init__(encoder, score_weight=score_weight)
+        self.loss_reduction = "mean"
+
+    def calculate_score(self, input_repr, target_repr, ctx):
+        score_correction_factor = 1 / input_repr.size()[0]
```

Comment:
This correction should only happen `if impl_params`, correct? If yes, I would pass `impl_params` to the constructor and set a variable like `self.double_batch_size_mean = impl_params`. Within in `calculate_score` you could than do:

```python
score = super().calculate_score(input_repr, target_repr)
if not self.double_batch_size_mean:
    return score

...
```